source/val/validate_extensions.cpp

```diff
@@ -1082,6 +1082,23 @@ spv_result_t ValidateExtInstImport(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateExtInstWithForwardRefs(ValidationState_t& _,
+                                            const Instruction* inst) {
+  if (!_.HasExtension(kSPV_KHR_relaxed_extended_instruction)) {
+    return _.diag(SPV_ERROR_INVALID_DATA, inst)
+           << "OpExtInstWithForwardRefs instruction requires "
+           << "SPV_KHR_relaxed_extended_instruction";
+  }
+
+  if (!_.HasExtension(kSPV_KHR_non_semantic_info)) {
```

Comment:
SPV_KHR_non_semantic_info was made core in SPIR-V 1.6 so I'm not sure you'd see the extension in 1.6 binary versions, but I also this the parser should be able to check this.